CONTRIBUTING.md

```diff
@@ -1,19 +1,65 @@
-Code formatting
-===============
+# How to contribute
 
-Go provides a tool, `go fmt`, which facilitates a standardised format to Go 
-source code. SQLair has two additional policies:
+Thanks for your interest in SQLair! Contributions like yours make good projects
+great.
 
-Imports
--------
+When contributing, you must abide by the [Ubuntu Code of
+Conduct](https://ubuntu.com/community/ethos/code-of-conduct).
 
-Import statements are grouped into three sections: 
-- standard library 
-- third party libraries 
-- SQLair imports 
+## Licence and copyright
 
-The tool "go fmt" can be used to ensure each group is alphabetically sorted, 
-for example:
+By default, all contributions to SQLair are made under the Apache 2.0 licence.
+See the [licence](LICENSE) in the SQLair GitHub repository for details.
+
+All contributors must sign the [Canonical contributor licence
+agreement](https://ubuntu.com/legal/contributors), which grants Canonical
+permission to use the contributions. The author of a change remains the
+copyright owner of their code (no copyright assignment occurs).
+
+## Environment setup
+
+SQLair is written in Go version 1.18. This choice was made to maximize
+compatibility.
+
+## Building and testing
+
+Build SQLair with `go build` and test SQLair with `go test ./...`. It's as
+simple as that.
+
+SQLair uses [go check](https://pkg.go.dev/gopkg.in/check.v1?utm_source=godoc).
+To run a subset of tests, pass the `-check.f` flag to `go test` with a regex of
+the test names you wish to run.
+E.g.
+```go
+go test ./... -check.f TestExpr
+```
+
+# Code contributions
```

Comment:
This should be ## (as in, it's a section in the same # How to contribute doc)

Reply:
Nested this and everything else below